### PR TITLE
Element Router with string name

### PR DESCRIPTION
**Pull request by blankse**
It is not possible to set an route with an object as name. We must use a string and give the object as parameter.

Before:
`{{ url(document) }}`
`{{ url(object) }}`

After:
`{{ url('pimcore_element', {'element': document}) }}`
`{{ url('pimcore_element', {'element': object}) }}`

### CODE REVIEW

**Comment by brusch** [comment]
But this would break existing `{{ url(document) }}` code, right? 🤔

**Comment by blankse** [comment]
@brusch This code doesn't work anymore. Symfony 5 has a Typehint string on the name param. So it is only the fullpath string and the router dosn't match.

**Comment by dpfaffenbauer** [comment]
1. It should either resolve the full path
2. or it is a bc break: you cannot break what is already broken right? :P

File: lib/Routing/Element/Router.php [hunk 1]
```diff
@@ -75,16 +75,17 @@ public function getContext()
      */
     public function supports($name)
```
Comment (jdreesen) [inline lib/Routing/Element/Router.php:76]:
```suggestion
    public function supports(string $name)
```

**Comment by brusch** [comment]
@blankse I see, then it's fine of course 😊

File: lib/Routing/Element/Router.php [hunk 2]
```diff
@@ -73,18 +73,19 @@ public function getContext()
     /**
      * {@inheritdoc}
      */
-    public function supports($name)
+    public function supports(string $name)
```
Comment (brusch) [inline lib/Routing/Element/Router.php]:
```suggestion
    public function supports($name)
```

File: lib/Routing/Element/Router.php (hunk 1 above)
Comment (brusch) [reply lib/Routing/Element/Router.php:76]: Unfortunately this breaks the code: 
Declaration of Pimcore\Routing\Element\Router::supports(string $name) must be compatible with Symfony\Cmf\Component\Routing\VersatileGeneratorInterface::supports($name)

**Pull request merged**